### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by wow-such-code**
- remove dead code
- return null if experiment is not found
- add function to fetch property types of ALL entity types
- make fetching of children and parent samples more stable
- add mock for testing
- some tests

### CODE REVIEW

**Comment by jenniferboedker** [comment]
How can I verify if the new code works? Do you have a project where this code is already is included?

**Comment by wow-such-code** [comment]
added some tests

File: pom.xml [hunk 1]
```diff
@@ -1,64 +1,64 @@
...
+		</dependency>
+		<!-- https://mvnrepository.com/artifact/com.google.truth/truth -->
+		<dependency>
+			<groupId>com.google.truth</groupId>
+			<artifactId>truth</artifactId>
+			<version>1.0.1</version>
+			<scope>test</scope>
+		</dependency>
+	</dependencies>
 </project>
```
Comment (KochTobi) [inline pom.xml:64]: was this changed?

Comment (wow-such-code) [reply pom.xml:64]: no, just formatted after merge with previous PR :/

File: src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+public class TestOpenBisClient {
+
+  IOpenBisClient openbis;
+  OpenBisClientHelper helper;
+
+  @BeforeClass
+  public static void setUpBeforeClass() {}
+
+  @AfterClass
+  public static void tearDownAfterClass() {}
```
Comment (jenniferboedker) [inline src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java]: Why is this included? Is this required to let the tests run?

File: src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+  @Before
+  public void setUp() {
+    openbis = new OpenBisClientMock();
+  }
+
+  @After
+  public void tearDown() {}
```
Comment (jenniferboedker) [inline src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java]: Also this method, there happens nothing?

File: src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java (hunk 2 above)
Comment (wow-such-code) [reply src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java]: I was not sure, might be an instance of cargo cult programming.

Comment (wow-such-code) [reply src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java]: In any case they can be used if we ever set up some kind of testing system that needs initialisation and shutdown. Maybe we can remove them until then, though?

Comment (jenniferboedker) [reply src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java]: I would remove them 
```suggestion
  
```

File: src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java (hunk 3 above)
Comment (jenniferboedker) [reply src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java]:
```suggestion
```

File: src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java (hunk 2 above)
Comment (jenniferboedker) [reply src/test/java/life/qbic/openbis/openbisclient/TestOpenBisClient.java]: If you want to keep them its also fine for me :)

**Review by jenniferboedker — DISMISSED** [review]
LGTM

File: src/main/java/life/qbic/openbis/openbisclient/helper/OpenBisClientHelper.java [hunk 4]
```diff
@@ -40,7 +45,7 @@ public static ProjectFetchOptions fetchProjectsCompletely() {
     projectFetchOptions.withSpace();
     projectFetchOptions.withExperiments();
     projectFetchOptions.withLeader();
-    //TODO Samples could not be fetched
+    // TODO Samples could not be fetched
```
Comment (KochTobi) [inline src/main/java/life/qbic/openbis/openbisclient/helper/OpenBisClientHelper.java:48]: NIT: is this still relevant?

File: src/main/java/life/qbic/openbis/openbisclient/helper/OpenBisClientHelper.java [hunk 5]
```diff
@@ -99,4 +104,12 @@ public static ExperimentTypeFetchOptions fetchExperimentTypesCompletely() {
     return experimentTypeFetchOptions;
   }
 
+  public static List<PropertyType> getPropertiesOfEntityType(IEntityType type) {
+  List<PropertyType> res = new ArrayList<>();
+  List<PropertyAssignment> assignments = type.getPropertyAssignments();
+  for (PropertyAssignment as : assignments) {
+    res.add(as.getPropertyType());
+  }
```
Comment (KochTobi) [inline src/main/java/life/qbic/openbis/openbisclient/helper/OpenBisClientHelper.java]:
```suggestion
  for (PropertyAssignment propertyAssignment : assignments) {
    res.add(propertyAssignment.getPropertyType());
  }
```
I would avoid using an English word such as `as`

**Review by KochTobi — DISMISSED** [review]
LGTM
Nit: change the variable name `as` to something else.

File: src/main/java/life/qbic/openbis/openbisclient/helper/OpenBisClientHelper.java (hunk 4 above)
Comment (wow-such-code) [reply src/main/java/life/qbic/openbis/openbisclient/helper/OpenBisClientHelper.java:48]: I don't know, not my TODO afaik

**Review by KochTobi — APPROVED** [review]
LGTM

File: pom.xml (hunk 1 above)
Comment (KochTobi) [reply pom.xml:64]: we figure that this is due to tabs vs. spaces
